### PR TITLE
[libc] fix HashTable warnings and build problems

### PR DESCRIPTION
**Pull request by SchrodingerZhu**
According to https://lab.llvm.org/buildbot/#/builders/163/builds/48002, the generic build on HashTable fails with two major issues with `werror`:
1. warnings on `error: suggest braces around initialization of subobject`.
2. `__support/HashTable` tests are built regardless of its entrypoints`

This PR attempts to fix such issues.

### CODE REVIEW

File: libc/test/src/__support/CMakeLists.txt [hunk 1]
```diff
@@ -181,4 +181,12 @@ add_subdirectory(File)
 add_subdirectory(RPC)
 add_subdirectory(OSUtil)
 add_subdirectory(FPUtil)
-add_subdirectory(HashTable)
+
```
Comment (jhuber6) [inline libc/test/src/__support/CMakeLists.txt:184]: The tests are build after all the sources right? It should be easier to just check the target like `if(TARGET libc.src.<whatever>.hsearch)`

File: libc/src/__support/HashTable/bitmask.h [hunk 2]
```diff
@@ -73,7 +73,7 @@ template <class BitMask> struct IteratableBitMaskAdaptor : public BitMask {
     return *this;
   }
   LIBC_INLINE IteratableBitMaskAdaptor begin() { return *this; }
-  LIBC_INLINE IteratableBitMaskAdaptor end() { return {0}; }
+  LIBC_INLINE IteratableBitMaskAdaptor end() { return {{0}}; }
```
Comment (jhuber6) [inline libc/src/__support/HashTable/bitmask.h:76]: Is this really the best way to silence the warning? Can we not explicitly default construct the type to make it clearer what's actually being done here?

**Comment by nickdesaulniers** [comment]
Please put the diagnostic observed from the compiler in the PR description.

**Review by michaelrj-google — APPROVED** [review]
LGTM

**Review by jhuber6 — APPROVED** [review]

**Comment by SchrodingerZhu** [comment]
Please remember to use the updated PR description when merging this. Sometimes github don't automatically draw the updates into the default commit message.

**Pull request merged**